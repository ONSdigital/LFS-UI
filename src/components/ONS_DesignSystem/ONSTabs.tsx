import React from "react";

interface Props {
    label?: string
    onClick: Function,
    items: any[]
}

export const ONSTabs = (props: Props) => (
    <div className={"tabs"}>
        {
            props.label !== undefined &&
            <label className="label">{props.label}</label>
        }
        <ul className="tabs__list tabs__list--row" role="tablist">
            {
                props.items.map((item: any, index: number) =>
                    <li key={index} id={"tabIdItem" + index} className="tab__list-item tab__list-item--row">
                        <a href={"#tabId"} onClick={() => props.onClick(item.name)}
                           className={"tab tab--row " + (item.active ? "tab--selected" : "")}
                           aria-selected={item.active ? "true" : "false"}
                           role="tab" tabIndex={1}>{item.name}</a>
                    </li>
                )
            }
        </ul>
    </div>
);
