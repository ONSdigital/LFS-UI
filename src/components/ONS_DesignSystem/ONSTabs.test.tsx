import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSTabs} from "./ONSTabs";
import Adapter from "enzyme-adapter-react-16";
import sinon from "sinon";

describe("ONS Tabs Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const tabsSelection = [
        {name: "Tab 1", active: true},
        {name: "Tab 2", active: false}
    ];
    
    const Props = {
        label: "I'm not your friend buddy",
        onClick: sinon.spy(),
        items: tabsSelection
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSTabs
                {...props}>
            </ONSTabs>
        )
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, Props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("should simulate a click event", () => {
        wrapper(mount, Props).find("a.tab--selected").simulate("click");
        expect(Props.onClick).toHaveProperty("callCount", 1);
    })
})