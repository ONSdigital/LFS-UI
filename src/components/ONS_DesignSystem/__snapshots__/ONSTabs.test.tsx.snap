// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Tabs Test matches Snapshot 1`] = `ShallowWrapper {}`;
